x/incentives/keeper/gauge.go

```diff
@@ -152,6 +152,10 @@ func (k Keeper) AddToGaugeRewards(ctx sdk.Context, owner sdk.AccAddress, coins s
 	if gauge.IsFinishedGauge(ctx.BlockTime()) {
 		return errors.New("gauge is already completed")
 	}
+
+	// Fixed gas consumption adding reward to gauges based on the number of coins to add
+	ctx.GasMeter().ConsumeGas(uint64(types.BaseGasFeeForAddRewardToGauge*len(coins)), "scaling gas cost for adding to gauge rewards")
```

Comment:
Just realized that this should actually be charging `baseAmount * (len(gauge.Coins) + len(coins)` since we allow for adding multiple coins at once here. The current implementation only scales by the number of coins being added, not the ones that already exist.

Reply:
i see, i originally thought it was just for new denoms to add. I'll change it to accomodate existing denoms